src/instructlab/sdg/datamixing.py

```diff
@@ -374,7 +374,70 @@ def _conv_pretrain(rec):
     return rec
 
 
-def _create_phase10_ds(generated_dataset: Dataset):
+def _create_auxiliary_dataset(
+    generated_dataset: Dataset, auxiliary_inst: Optional[Dict[str, List[str]]]
+):
+    # Samples that went through the auxiliary generation pipeline will
+    # have a dataset_type column created by that pipeline. If that's
+    # not present, then we may be running in a pipeline without any
+    # auxiliary dataset generation enabled.
+    if "dataset_type" not in generated_dataset.column_names:
+        return None
+    # If we didn't find any auxiliary instructions to load, then
+    # that's also another sign that we're not running with any
+    # auxiliary datasets enabled.
+    if auxiliary_inst is None:
+        return None
+    # This "base_document" dataset_type is set in the knowledge
+    # pipeline config, and represents samples that do not have the
+    # auxiliary generated document attached, so we filter those out.
+    auxiliary_ds = generated_dataset.filter(
+        lambda x: x["dataset_type"] != "base_document"
+    )
+    unique_document_auxiliary = (
+        auxiliary_ds.to_pandas()
+        .drop_duplicates(subset=["document"])
+        .reset_index(drop=True)
+    )
+    unique_document_auxiliary = Dataset.from_pandas(unique_document_auxiliary)
```

Comment:
Use `instructlab.sdg.utils.pandas.dataset_from_pandas_dataframe()`

Reply:
Yes, updated to do that now that that helper method merged. Thanks!